src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -273,6 +274,33 @@ def fetch_schedules(self, db: sqlite3.Connection, path: str) -> List[sqlite3.Row
                     if self._is_allowed_repeat(r, path)][0:1]
             return rows
 
+    def delete_references_to_unavailable_fs(self, available_fs_names: Set[str]) -> None:
+        timers_to_remove: Set[Tuple[str,str]] = set()
+        fs_to_remove: Set[str] = set()
+        self.timers_lock.acquire()
+        for key in iter(self.active_timers):
+            fs = key[0]
+            if fs not in available_fs_names:
+                timers_to_remove.add(key)
+                fs_to_remove.add(fs)
+                log.debug(f'Will remove timers for fs "{key}"')
+
+        for i in timers_to_remove:
```

Comment:
Why not just cancel the timer in the above loop?

Reply:
This approach is related to iterator invalidation.
Its not correct programming practice to delete the same element in the container that is being observed.

Reply:
oh, I mean you could do a list(self.active_timers.keys()) and then in a single iteration cancel timers.

Reply:
Updated the code with the suggestion.